Source/WebKit/UIProcess/API/Swift/WebPage+Configuration.swift

```diff
@@ -63,6 +65,26 @@ extension WebPage_v0 {
     }
 }
 
+extension WebPage_v0 {
+    @_spi(Private)
+    public struct DeviceSensorAuthorization {
+        public enum Permission: Hashable, Sendable {
+            case deviceOrientationAndMotion
+            case mediaCapture(WKMediaCaptureType)
+        }
+
+        let decision: (Permission, WebPage_v0.FrameInfo, WKSecurityOrigin) async -> WKPermissionDecision
```

Comment:
For a block I’d probably give it a more action-y name, like `decisionHandler` to clarify it isn’t a decision in itself. And then ditto as the argument name.